### PR TITLE
unit strings rounding todo

### PR DESCRIPTION
**Pull request by NeuralSpaz**
changes the way physic units represented as string.
implement rounding for nanoAsString and microAsString.
add rounding test cases for nanoAsString.
add rounding test cases for microAsString.
fix examples for unit rounding.

### CODE REVIEW

File: conn/physic/units_test.go [hunk 1]
```diff
@@ -228,32 +271,42 @@ func TestMicroAsString(t *testing.T) {
 		{-1000000, "-1"},
 		{1100000, "1.100"},
```
[the remark this answers is not in the document]
Comment (NeuralSpaz) [reply conn/physic/units_test.go]: yes, 1000501 gives 100%

**Comment by maruel** [comment]
gohci

**Comment by maruel** [comment]
I've confirmed with `go test -bench=.` that there is no significant performance change even with the additional conditions.

**Comment by maruel** [comment]
Thanks a lot for this excellent PR!

**Comment by NeuralSpaz** [comment]
Thank you for all your help with this, #294 and #292  :)
It is a great a project, and happy I can contribute to it.

**Pull request merged**